enum.rb

```diff
@@ -0,0 +1,521 @@
+  def my_select
+    arr = []
+    unless block_given?
+      return to_enum :my_select, print('No block was given.', "\n", "\n")
+    end
+
+    if !empty?
+      0.upto(length - 1) do |index|
+        arr.push(self[index]) if yield(self[index])
+      end
+    else
+      print '=> ', 'Array is empty.', "\n", "\n"
+    end
+    print('=> ', arr, "\n", "\n") unless arr.empty?
+    arr
+  end
+
+  # rubocop: enable Metrics/MethodLength
+  # *********************************
+  # End of my_select method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_all method.
+  # *********************************
+  # rubocop: disable Style/CaseEquality, Layout/EndAlignment
+  def my_all?(arg = nil)
+    temp = true
+    0.upto(length - 1) do |index|
+      temp &&= if block_given?
+                 yield(self[index])
+               else
+                 (arg === self[index])
+        end
+    end
+    temp
+  end
+
+  # rubocop: enable Style/CaseEquality, Layout/EndAlignment
+  # *********************************
+  # End of my_all? method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_any? method.
+  # *********************************
+  # rubocop: disable Style/CaseEquality, Layout/EndAlignment
+  def my_any?(arg = nil)
+    temp = false
+    0.upto(length - 1) do |index|
+      temp ||= if block_given?
+                 yield(self[index])
+               else
+                 (arg === self[index])
+        end
+    end
+    temp
+  end
+
+  # rubocop: enable Style/CaseEquality, Layout/EndAlignment
+  # *********************************
+  # End of my_any? method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_none? method.
+  # *********************************
+  # rubocop: disable Style/CaseEquality, Layout/EndAlignment
+  def my_none?(pat = nil)
+    temp = true
+    0.upto(length - 1) do |index|
+      temp &&= if block_given?
+                 !yield(self[index])
+               else
+                 !(pat === self[index])
+        end
+    end
+    temp
+  end
+
+  # rubocop: enable Style/CaseEquality, Layout/EndAlignment
+  # *********************************
+  # End of my_none? method.
+  # *********************************
+
+  # *************************(********
+  # Beginning of my_count method.
+  # *********************************
+  # rubocop: disable Style/CaseEquality
+  NOTHING = Object.new
+
+  def my_count(arg = NOTHING)
+    if block_given?
+      my_select { |index| yield index }.length
+    else
+      my_select { |index| index == arg || NOTHING === arg }.length
+    end
+  end
+
+  # rubocop: enable Style/CaseEquality
+
+  # *********************************
+  # End of my_count method.
+  # *********************************
+
+  # *********************************
+  # Beginning of my_map method.
+  # *********************************
+  def my_map
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns a new array with the results of running block once for every element in enum.
```ruby
array = Array.new(100){rand(0...9)}
array.my_map(&block) == array.map(&block) # should return true
```

Reply:
Tests run satisfactorily.